File: CSharp/ByTier/B/2139_B_Cake_Collection.cs
// Problem: CF 2139 B - Cake Collection
// https://codeforces.com/contest/2139/problem/B

/*
B. Cake Collection
Algorithms/Techniques: Greedy algorithm with sorting

Time Complexity: O(n log n + n) per test case, where n is the number of ovens.
Space Complexity: O(n) for storing the array of cake rates.

The problem involves maximizing the number of cakes collected over m seconds,
where each oven produces a_i cakes per second. At each second, Maple can switch
to any oven and collect all accumulated cakes from that oven. This is a greedy
problem where we want to collect from the oven with the highest production rate
at each step.

Approach:
1. Sort ovens in descending order of cake production rate.
2. For the first 'm' seconds, take the top 'min(n, m)' ovens and accumulate
   their contributions considering how many seconds have passed (multiplier).
3. The maximum total cakes is computed by summing (a[i] * (m - i)) for i from 0 to min(n, m) - 1.
*/

using System;
using System.Collections.Generic;
class Program {
  static void Main() {
    int t = int.Parse(Console.ReadLine());
    while (t-- > 0) {
      var line = Console.ReadLine().Split();
      int n = int.Parse(line[0]);
      long m = long.Parse(line[1]);
      var arr = Array.ConvertAll(Console.ReadLine().Split(), long.Parse);
      // Sort ovens in descending order of cake production rate
      Array.Sort(arr, (x, y) => y.CompareTo(x));
      // Determine how many ovens to consider (up to m or n)
      long take = Math.Min(n, (int)m);
      long total = 0;
      // Accumulate maximum cakes by taking from highest production rate ovens
      for (int i = 0; i < take; i++) {
        total += arr[i] * (m - i);
      }
      Console.WriteLine(total);
    }
  }
}


// https://github.com/VaHiX/codeForces/

File: Java/ByTier/B/2139_B_Cake_Collection.java
// Problem: CF 2139 B - Cake Collection
// https://codeforces.com/contest/2139/problem/B

/*
B. Cake Collection
Algorithms/Techniques: Greedy approach with sorting
Time Complexity: O(n log n + n) per test case due to sorting and single pass
Space Complexity: O(1) excluding input storage

Problem Summary:
Maple has n ovens, each baking a[i] cakes per second. She can collect cakes from any oven at the end of each second.
She wants to maximize total cakes collected in m seconds.

Approach:
- Sort the ovens by cake production rate (ascending).
- For each second from the end (m-1 down to 0), we decide which oven to visit.
- To maximize, at each step, choose the oven with highest remaining capacity.
- Since after m seconds all ovens will have baked a[i]*m cakes,
  but we only collect once per oven, use greedy selection in reverse:
    - Visit ovens with highest rates first.
    - Accumulate result by multiplying rate with time left.

*/

import java.util.*;
public class B_Cake_Collection {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0) {
            int n = sc.nextInt(), m = sc.nextInt();
            int[] a = new int[n];
            for(int i=0; i<n; i++) a[i] = sc.nextInt();
            Arrays.sort(a); // Sort ovens by rate (ascending)
            long result = 0;
            // Traverse from highest rate oven to lowest
            for(int i=n-1; i>=0; i--) {
                // Calculate how many seconds left to use this oven (from m down to current position)
                int timeLeft = m - (n - 1 - i); 
                if(timeLeft > 0) {
                    result += (long) a[i] * timeLeft;
                }
            }
            System.out.println(result);
        }
        sc.close();
    }
}


// https://github.com/VaHiX/codeForces/